### ISSUE TITLE
EPG improvements

### ISSUE THREAD

**Issue by madsci** [xbmc/xbmc#16974]
### Describe the bug
Here is a clear and concise description of what the problem is:

I am using KODI Leia (v18.3, LibreElec) with PVR.Zattoo addon daily and I am happy with the media station. Nevertheless, 2 improvements would make the experience even better:

- I am often using reply-TV (7 days history). The corresponding broadcast is selected by the EPG. When pressing "stop" or "back" to return back to EPG, the time/date is reset to now which is annoying. I would appreciate if it jumps to the just played entry.
- The second is also about reply-TV/EPG function. Playing is automatically stopped when the selected broadcast (with some extra time to compensate programm shifts) ends. Would it be possible to continue with the next broadcast? Furthermore, the EPG jumps back to now.

I have also posted at PVR.zattoo addon repository but was forwarded to KODI repository.

https://github.com/rbuehlma/pvr.zattoo/issues/60

Thank you for considering my input

Here is a clear and concise description of what the problem is:
When using PVR.zattoo addon and reply-TV (7 days history), the EPG always jups to current date when returing to EPG.  Furthermore, playback ends when broadcast in reply mode has finished.

## Expected Behavior
Here is a clear and concise description of what was expected to happen:

- In EPG return to last played item especially in reply-TV mode
- Seamless playing of next broadcast when one ends/finishes

## Actual Behavior

When using PVR.zattoo addon and reply-TV (7 days history), the EPG always jups to current date when returing to EPG.  Furthermore, playback ends when broadcast in reply mode has finished.

## Possible Fix

Improve EPG frontend

### To Reproduce

1. The situation happens every time when playback ist interrupted and returning back to EPG view.
2. Select old EPG broadcast, press OK and watch
3. return to EPG (by back or stop) and focus jumps to current time/date

### Debuglog

The debuglog can be found here:
No debug log available. The situation is a missing feature.

### Screenshots 
Here are some links or screenshots to help explain the problem:

## Additional context or screenshots (if appropriate)
Here is some additional context or explanation that might help:

### Your Environment
Used Operating system:

 - [ ] Android
 - [ ] iOS
 - [x] Linux
 - [ ] OSX
 - [ ] Raspberry-Pi
 - [ ] Windows
 - [ ] Windows UWP

 - Operating system version/name: LibreElec 9.1.005
 - Kodi version: Leia v18.3

*note: Once the issue is made we require you to update it with new information or Kodi versions should that be required.
Team Kodi will consider your problem report however, we will not make any promises the problem will be solved.*

**Comment by madsci** [comment]
I have updated the issue. Please remove ignore rules tag.
Thank you

**Comment by ksooo** [comment]
As it works as designed, we're not facing a bug here, but improvements/feature requests.

**Comment by madsci** [comment]
Thank you ksooo for checking. Will my request go directly into official feature request path or where should I address the request to?
In the mean time, I have seen that the request is actually pretty old. Already 2.5 years ago, rbuehlma (developer of Zattoo PVR) brought up excactly this topic (https://forum.kodi.tv/showthread.php?tid=316999&page=2&highlight=replay). How is the status?

I would say close to 2020, no one does deny anymore that Video-on-Demand and Reply-TV is the future and KODI should implement such a feature in a professional and user-friendly way.

**Comment by ksooo** [comment]
For sure this is a valid feature request. And when some developer with the required skills will be in the mood to implement this it will eventually become reality. If not, we will have to wait. That's how open source projects driven by 100% unpaid volunteers do work.

**Comment by tiphi5** [comment]
Hello,
I am a new user to Kodi 19 and have noticed this exact behavior as well. I use it with Ubuntu 18.04 and have also tried it with Windows 10. The same behavior everywhere.
When I read in the news about the latest version "Fixed: playing archived programs not selected when opening the Guide window" I installed the latest version 19.1 and thought it was fixed now, but unfortunately not. 
I use the Teleboy PVR addon. 
If you play a past program in the guide and you play the stream to the very end, the focus in the guide afterwards is not on the just played past program, but the focus points to the current time. If the show was longer than the stream was recorded, you have to search for the show in the guide again and play the next show.
It would be nice, of course, if the next program would be played automatically without interruption.
Is there really no one here who has programmed this feature?

Translated with www.DeepL.com/Translator (free version)

**Comment by ksooo** [comment]
> Fixed: **playing** archived programs not selected when opening the Guide window" I installed the latest version 19.1 and thought it was fixed now, but unfortunately not.

 Well, it is. You misinterpreted what the fix is for - "playing" is different from "last played".

**Comment by ksooo** [comment]
Last comment deleted, because I was probably wrong with that statement.

**Comment by ksooo** [comment]
> the focus in the guide **afterwards**

 Does this mean, after playback of the past program stopped?

**Comment by tiphi5** [comment]
Yes, it does. I think, because the stream ends. And the last window is called, in this case the Guide.

**Comment by ksooo** [comment]
This is not what the fix mentioned in 19.1. release notes is supposed to fix. It fixes the Guide window not selecting the **currently playing** past program when openend. Selecting the last played program on open is a feature not yet implemented.

**Comment by tiphi5** [comment]
OK, I understand. Then I hope this feature is on the to-do-list.
Thanks
